### PR TITLE
rotationFromMatrix-fix

### PR DESCRIPTION
**Pull request by Simon-Steinmann**
Where I found the issue discussed and fixed here:
https://github.com/cyberbotics/webots/pull/2155

I implemented a conversion method from
https://rock-learning.github.io/pytransform3d/_modules/pytransform3d/rotations.html#axis_angle_from_matrix

### CODE REVIEW

File: urdf2webots/math_utils.py [hunk 1]
```diff
@@ -97,24 +97,42 @@ def matrixFromRotation(rotation):
...
-        else:
-            cosAngle = 1.0
-
-    rotation.append(matrix[7] - matrix[5])
-    rotation.append(matrix[2] - matrix[6])
-    rotation.append(matrix[3] - matrix[1])
-    rotation.append(math.acos(cosAngle))
-    return rotation
+def rotationFromMatrix(R):
+    R = numpy.array(R).reshape(3,3)
```
Comment (Simon-Steinmann) [inline urdf2webots/math_utils.py:101]: I added this line, so we can either parse a 3x3 matrix, or a dim-1 len(9) list

File: urdf2webots/math_utils.py [hunk 2]
```diff
@@ -97,24 +97,43 @@ def matrixFromRotation(rotation):
...
+        # The threshold is a result from this discussion:
+        # https://github.com/rock-learning/pytransform3d/issues/43
+        # The standard formula becomes numerically unstable, however,
+        # Rodrigues' formula reduces to R = I + 2 (ee^T - I), with the
+        # rotation axis e, that is, ee^T = 0.5 * (R + I) and we can find the
+        # squared values of the rotation axis on the diagonal of this matrix.
+        # We can still use the original formula to reconstruct the signs of
+        # the rotation axis correctly.
+        sign = numpy.sign(axis_unnormalized)
+        a[:3] = numpy.sqrt(0.5 * (numpy.diag(R) + 1.0)) * numpy.where(sign == 0, 1, sign)
```
Comment (Simon-Steinmann) [inline urdf2webots/math_utils.py:126]: 125 and 126 are changed, because sign() can return zero. The value must be 1 or -1

**Review by omichel — CHANGES_REQUESTED** [review]

File: urdf2webots/math_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Simon-Steinmann) [reply urdf2webots/math_utils.py:127]: Like I mentioned, I took it from the link I posted, and it works.

[the remark this answers is not in the document]
Comment (omichel) [reply urdf2webots/math_utils.py:127]: OK, so let's keep it like that until we find a problem...

**Review by omichel — APPROVED** [review]
Thank you!

**Pull request merged**